src/components/base/BaseDatepicker.vue

```diff
@@ -0,0 +1,49 @@
+<script setup lang="ts">
+
+withDefaults(
+  defineProps<{
+    modelValue?: string;
+  }>(),
+  {
+    modelValue: "",
+  }
+);
+</script>
+
+<template>
+  <div class="date-picker">
+    <label for="" class="date-picker__label">
+      <slot />
+      <CalendarIcon class="date-picker__calendar" />
+      <input type="date" class="date-picker__input" @input="
+        $emit('update:modelValue', ($event.target as HTMLInputElement).value)
```

Comment:
You might need to also specify `defineEmits` in script tag

Reply:
same as above